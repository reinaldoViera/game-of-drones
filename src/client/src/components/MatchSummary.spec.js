import React from 'react';
import {
    mount
} from 'enzyme';
import {
    MatchSummary
} from "./MatchSummary";
import renderer from 'react-test-renderer';

const rounds = [{
    winner: 1,
    name: 'Player 1'
}, {
    winner: 2,
    name: 'Player 2'
}, {
    winner: null,
    name: null
}]

describe('Loading component', () => {
    test('should display rounds data', () => {
        const result = renderer.create( 
            <MatchSummary rounds = {rounds} classes = {{}}/>
        ).toJSON();
        expect(result).toMatchSnapshot();
    });
    test('should show winner view', () => {
        const result = renderer.create( 
            <MatchSummary winner={'Player 1'} rounds={rounds} classes = {{}}/>
        ).toJSON();
        expect(result).toMatchSnapshot();
    });
    test('should react to New Game Button', () => {
        const onNewGame = jest.fn();
        const el = mount( <MatchSummary onNewGame={onNewGame} winner={'Player 1'} rounds={rounds} classes = {{}}/>);
        el.find('Button').simulate('click');
        expect(onNewGame.mock.calls.length).toEqual(1);
    })
    
})