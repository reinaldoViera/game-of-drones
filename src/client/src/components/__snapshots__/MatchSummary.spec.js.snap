// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading component should display rounds data 1`] = `
<div
  className="MuiPaper-root-1 MuiPaper-elevation5-8 MuiPaper-rounded-2"
>
  <h3
    className="MuiTypography-root-28 MuiTypography-h5-44"
  >
    Score
  </h3>
  <div
    className="MuiTypography-root-28 MuiTypography-body1-37"
  >
    <table
      id="summary-table"
    >
      <tbody>
        <tr>
          <th>
            Round
          </th>
          <th>
            Winner
          </th>
        </tr>
        <tr>
          <td>
            0
          </td>
          <td>
            Player 1
          </td>
        </tr>
        <tr>
          <td>
            1
          </td>
          <td>
            Player 2
          </td>
        </tr>
        <tr>
          <td>
            2
          </td>
          <td>
            ---
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`Loading component should show winner view 1`] = `
<div
  className="MuiPaper-root-1 MuiPaper-elevation5-8 MuiPaper-rounded-2"
>
  <h3
    className="MuiTypography-root-28 MuiTypography-h5-44"
  >
    Player 1 is the MASTER!!!
  </h3>
  <h3
    className="MuiTypography-root-28 MuiTypography-h5-44"
  >
    Score
  </h3>
  <div
    className="MuiTypography-root-28 MuiTypography-body1-37"
  >
    <table
      id="summary-table"
    >
      <tbody>
        <tr>
          <th>
            Round
          </th>
          <th>
            Winner
          </th>
        </tr>
        <tr>
          <td>
            0
          </td>
          <td>
            Player 1
          </td>
        </tr>
        <tr>
          <td>
            1
          </td>
          <td>
            Player 2
          </td>
        </tr>
        <tr>
          <td>
            2
          </td>
          <td>
            ---
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <button
    className="MuiButtonBase-root-89 MuiButton-root-63 MuiButton-contained-74 MuiButton-containedPrimary-75 MuiButton-raised-77 MuiButton-raisedPrimary-78"
    disabled={false}
    id="new-game-btn"
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiButton-label-64"
    >
      New Game
    </span>
    <span
      className="MuiTouchRipple-root-92"
    />
  </button>
</div>
`;
